src/Tracking/filter_functions.jl

```diff
@@ -0,0 +1,218 @@
+
+Compute the psi-s correlation between a floe and a dataframe of candidate floes. Adds the 
+psi-s correlation,  psi-s correlation score (1 - correlation), and the result of the threshold function
+to the columns of `candidates`.
+""" 
+@kwdef struct PsiSCorrelationThresholdFilter <: AbstractFloeFilterFunction
+    area_variable = :area
+    threshold_column = :psi_s_correlation_test
+    threshold_function = PiecewiseLinearThresholdFunction(100, 800, 0.14, 0.1)
+end
+
+#TODO: Add option to include the confidence intervals with the normalized cross correlation tests.
+function (f::PsiSCorrelationThresholdFilter)(floe, candidates, _::Val{:raw})
+    
+
+    rfloe(p2) = round(normalized_cross_correlation(floe.psi, p2), digits=3)
+    transform!(candidates,  [:psi] => ByRow(rfloe) => :psi_s_correlation)
+    candidates[!, :psi_s_correlation_score] = 1 .- candidates[!, :psi_s_correlation]
+
+    # Future work: add computation of the confidence intervals for psi-s corr here.
+    transform!(candidates, [f.area_variable, :psi_s_correlation_score] =>
+        ByRow(f.threshold_function) => f.threshold_column
+    )
+end
+
+"""
+    ChainedFilterFunction(filters::Vector)
+
+Compute the psi-s correlation between a floe and a dataframe of candidate floes. Adds the 
+psi-s correlation,  psi-s correlation score (1 - correlation), and the result of the threshold function
+to the columns of `candidates`. The input is a vector of functions that take a floe and a candidates data
+frame as inputs. The default is to use seven filter functions with the threshold functions as defined
+in Watkins et al. 2026. These filters are:
+    1. DistanceThresholdFilter
+    2-5. RelativeErrorThresholdFilters for area, convex_area, major_axis_length, and minor_axis_length
+    6. ShapeDifferenceThresholdFilter
+    7. PsiSCorrelationThresholdFilter
+Filters 2-7 use a PiecewiseLinearThresholdFunction.
+
+""" 
+@kwdef struct ChainedFilterFunction <: AbstractFloeFilterFunction
+    filters = [
+        DistanceThresholdFilter(),
+        RelativeErrorThresholdFilter(variable=:area),
+        RelativeErrorThresholdFilter(variable=:convex_area),
+        RelativeErrorThresholdFilter(variable=:major_axis_length),
+        RelativeErrorThresholdFilter(variable=:minor_axis_length),
+        ShapeDifferenceThresholdFilter(),
+        PsiSCorrelationThresholdFilter()
+        ]
+end
+
+function (f::ChainedFilterFunction)(floe, candidates)
+    for filter_fun in f.filters
+        filter_fun(floe, candidates)
+    end
+end
+
+
+"""
+    LopezAcosta2019ChainedFilterFunction(floe, candidates)
+
+The LopezAcosta2019ChainedFilterFunction is a special case of ChainedFilterFunction
+with parameters and threshold functions set based on Lopez-Acosta et al. 2019. The set
+of threshold filters is the same as in the default ChainedFilterFunction, but using
+stepwise threshold functions instead of piecewise.
+"""
+LopezAcosta2019ChainedFilterFunction = ChainedFilterFunction(
```

Comment:
I like this example function. We can move it to the `LopezAcosta2019` submodule later,